enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,434 @@
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+Prerequisites:
+
+* The OpenStack cloud runs the [Bare Metal (ironic)][openstack-ironic] service
+  to provide baremetal management
+* The OpenStack [Compute (nova)][openstack-nova] and [Networking
+  (neutron)][openstack-neutron] services are configured to serve bare metal
+  nodes using the Nova API
+* All of the additional complexity inherent to baremetal is handled by the IaaS
+  cloud and its administrators:
+    * Enrolling new hardware and discovering its components
+    * Wiping the disks between deployments
+    * Booting the system up from the desired image
+    * Networking
+* The choice between deploying a virtual or baremetal machine is done by
+  selecting an appropriate Nova flavor and Neutron network
+* Both virtual and baremetal machines consume the same image from glance
+* The installer will have no insight into the Ironic networking
+    * All it will do is create a node with the user-specified flavor and
+      network
+
+We target four distinct use-cases:
+
+1. Install with BM workers on the installer-provisioned network
+2. Install with BM workers on a preexisting network
+3. Install with both VM and BM workers (BMs on a preexisting network)
+4. Install on VMs then add BM workers (BMs on a preexisting network)
+
+### Implementation work
+
+#### Case 1. Install with BM workers on the installer-provisioned network
+
+Booting BMs in a virtual network requires specific hardware; this case is a
+simplified one and is not expected to be the most represented in real-world
+installations.
+
+Prerequisites for the user:
+
+* Being able to provision BM nodes on a tenant network
+
+Expected work:
+
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+* Document how to use:
+  * document the requirements (specific hardware mentioned above)
+  * set the Ironic flavor in `install-config.yaml`
+
+#### Case 2. Install with BM workers on a preexisting network
+
+This case leverages [Provider Networks and Custom
+Subnets](openstack-customer-provided-subnets.md). The cluster is installed on
+the same pre-existing network where the BMs boot.
+
+Prerequisites for the user:
+
+* Being able to provision VMs nodes on the preexisting subnet where BMs run.
+
+Expected work:
+
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * set the appropriate Ironic-backed Nova flavor in `install-config.yaml`
+
+#### Case 3. Install with both VM and BM workers (BMs on a preexisting network)
+
```

Comment:
In this case, the user provisions two machine-pools: one VM-based, the other BM-based. It's two separate cases for us, because they require different implementation work. I tried to be more explicit with a new "description" section, please let me know if it's worse :sweat_smile: